File: java/src/main/java/site/markhenrick/recreational/adventofcode/y2015/Day01.java
package site.markhenrick.recreational.adventofcode.y2015;

import site.markhenrick.recreational.common.StringUtil;

@SuppressWarnings({"CharUsedInArithmeticContext", "StatementWithEmptyBody"})
class Day01 {

	public static long solvePart1(final CharSequence input) {
		return input.length() - 2 * StringUtil.charStream(input)
			.filter(character -> character == ')')
			.count();
	}

	public static Integer solvePart2(final CharSequence input) {
		var i = 0;
		for (var floor = 0; i < input.length() && floor >= 0; floor += 81 - 2 * input.charAt(i++)); // Real K&R hours
		return i;
	}
}

File: kotlin/src/main/kotlin/site/markhenrick/recreational/adventofcode/y2025/Day01.kt
package site.markhenrick.recreational.adventofcode.y2025

private const val MODULO = 100
private const val INITIAL = 50

object Day01 {
    fun solvePart1(input: String): Int = input.lineSequence()
        .map {
            assert(it[0] == 'L' || it[0] == 'R')
            val sign = if (it[0] == 'L') -1 else 1
            val value = it.substring(1).toInt()
            return@map sign * value
        }
        .runningFold(INITIAL) { a, b -> Math.floorMod(a + b, MODULO) }
        .filter { it == 0 }
        .count()
}
